### DISCUSSION TITLE
Test Ads are Not Working

### DISCUSSION THREAD

**Discussion by mga85** [Poing-Studios/Godot-AdMob-Android-iOS#102] [Q&A] [answered]
Hello,

AdMob Test ads are not loading in my game. I tried to implement the plugin by using default test ids provided by AdMob (which are already available in the plugin) and I  also updated Androidmanifest file with the id for testing. 
I get the AdMob Load Error 3. In AdMob Developer page I could not see any recent problem related to AdMob Servers. Somehow it does not find any proper test add to load. 
After testing in my application, I created another project based on Example.tscn in your plug in, this one also does not load any test ads.
Is there a problem with AdMob connection ?
I am using Godot 3.4.2 and also plugin 3.4

**Comment by gumaciel** [comment]
Hello, thank you for using our plugin and sorry for delay.

According to this link, the Error 3 mean this: https://support.google.com/admob/thread/2387222/ad-failed-to-load-3?hl=en&msgid=29667300

```
If you are getting this error, then your code is correct. The issue is that AdMob does not always have an ad to return for every request. This may happen particularly if you have just registered your AdMob publisher ID, as it takes some time and multiple requests before the new ID starts returning ads.
Another reason your fill rate may be low is that you don't have AdSense backfill enabled, or you have filtered out some ads. Check your app settings to see if that is the case.
```
Try to use on other phone or Simulator also try to use on Mobile Data.

This error happens only with Banner or all Ad Types(Interstitial, Rewarded)?

Watch this video tutorial: https://youtu.be/MW4hUR0d9lc

I will try to send you a `.apk` to you test soon

**Comment by mga85** [comment]
Hi,

Sorry for late reply. It was happening in all kinds of ads. But the response provided in the Google link above provided should not be related to the test ads, since they should push those dummy/test ads directly from their server and not waiting for some provider to fill the ads.
Weird thing is now I am using two mobile phones to test: one is HTC One M9 (Android 7.0) and the other one is Oneplus 8t (Android 11), now the ads are shown in Oneplus 8t, but not on HTC One M9. In HTC I still get the same error. What confuses me is I am using same test ad ids for both phones and literally running them at the same time. One shows the ads another does not. I asked another friend to test in a relatively new phone and in his case rewarded ads are loading but not the interstitial ads.

**Reply by gumaciel** [reply]
Hello, It may be that there are no test ads available at the moment.

if you are using UMP and you click "Do not Consent", under GDPR, the chance of not showing test ads is high, Try testing with an ad you created on https://admob.google.com/

**Comment by mga85** [comment, marked as answer]
Hi,

I am not using UMP. I tried to create a test add in Admob, but it failed since my app is not published yet and Admob status is always "review required".
I thought the ads from the page **https://developers.google.com/admob/android/test-ads** are always available.
I guess, I will check it when I publish the app.
I have a question, in Admob Editor of Plug-In under General Tab the option "Enabled", what does it stand for ? In Youtube video  there are both Real and Enabled options, but in 3.4, there is only "Enabled"

**Reply by gumaciel** [reply]
Hi, `is_real` was removed, read here: https://github.com/Poing-Studios/godot-admob-android/discussions/77#discussioncomment-2143195

And the `Enabled` can be used for if the user buys a paid version of the game, just deactivate "Enabled", in other words, it is to activate or deactivate the plugin.

Thinking about it now, it might be a little useless, because when exporting the game you can just uncheck the AdMob plugin, so it won't export with AdMob and the game will be free of ads

> I guess, I will check it when I publish the app.

I'm not sure, but you don't necessary need to publish your app to create Ads Ids and test it, it can take some time to they work, read here: https://support.google.com/admob/answer/9469204?hl=en

**Reply by mga85** [reply]
Hi, thanks a lot for your help. I will check with Admob then, regarding the review requirement.

**Reply by mga85** [reply]
Hi,

would it be possible to send me the test .apk you mentioned before ? I am still having the same issue. I checked AdMob website and they state demo adds are always filled. I tried Unpublished App approach in AdMob but it also failed.

https://support.google.com/admob/answer/9388275?hl=en

![google admob](https://user-images.githubusercontent.com/98719428/156175376-a4d5a7cb-14f4-4552-8c42-24e955e2a0ad.png)

**Reply by gumaciel** [reply]
hello, i will try to send to you soon

**Reply by gumaciel** [reply]
For some reason i can't upload the .zip here, i will try on other way

**Reply by gumaciel** [reply]
Here is the link: https://www.mediafire.com/file/2bhybax89kzy42k/AdMobAddon.apk/file

Build on Godot Version: v3.4.3 https://downloads.tuxfamily.org/godotengine/3.4.3/
Android Plugin Version: v2.0.0 https://github.com/Poing-Studios/godot-admob-android/releases/tag/v2.0.0
Editor Plugin Version: v1.2.4 https://github.com/Poing-Studios/godot-admob-editor/releases/tag/v1.2.4


**WARNING** This link will be available only for 14 days

**Reply by gumaciel** [reply]
Also: The Scene that i'm using is located on AdMob Editor Plugin on folder: `res://addons/admob/test/Example.tscn` and i'm didn't made any changes, i'm using the demo ids of Android: https://developers.google.com/admob/android/test-ads#demo_ad_units which is already in the editor when you download it

![Captura de Tela 2022-03-01 às 19 20 25](https://user-images.githubusercontent.com/20030153/156258619-79204a5e-7523-4833-8f03-dc59ef68003e.png)

**Reply by mga85** [reply]
Hello,

Thanks a lot, in my end only banner and interstitial ads worked after I run your apk. I also exported Example.tscn from the plugin and it runs banner and interstitial ads only. Somehow rewarded videos are not loading in production (under both WiFi and 4G connection). When I run the game in Android Studio or Xcode emulator, all the ads are loaded without any issue. In real devices(both Android and IOS), loading of rewarded test ads are inconsistent . They mostly fail in Android and sometimes in IOS.

Status when I launch the app in my phone
![admob](https://user-images.githubusercontent.com/98719428/156922320-719d48ff-3241-4fa8-8f2b-c3ad7fd191dd.jpg)

**Reply by gumaciel** [reply]
hello, check the Ids that Google provides for Demo Ids: 

https://developers.google.com/admob/ios/test-ads#demo_ad_units
https://developers.google.com/admob/android/test-ads#demo_ad_units

they may have changed, if everything is ok, please send the error code, if the error is "ERROR_CODE_NO_FILL", then that mean implantation is correct but there is no ads available in your area at the moment

try using a VPN or creating an ad yourself on admob.com, remember that it may take some time for ads to work if they are created on admob.com

read more here: https://support.google.com/admob/answer/9469204?hl=en

**Reply by mga85** [reply]
Hello,

The message I get is "ERROR_CODE_NO_FILL" and ids are correct. Regarding "there is no ads available in your area at the moment" as per previous Google screenshot above test ads are always filled (red square section). My feeling is Google somehow contradicts with its own statement and does not deliver the ads. However, I could not find any open discussions regarding this in AdMob Developer forums.

**Reply by mga85** [reply]
Hello, 

In production everything works as expected in Android. Apparently there is an issue in Google side for showing test ads.
Thanks for your helps so far.
